avm/res/insights/private-link-scope/main.bicep

```diff
@@ -0,0 +1,287 @@
+  name: string?
+
+  @description('Optional. The location to deploy the private endpoint to.')
+  location: string?
+
+  @description('Optional. The service (sub-) type to deploy the private endpoint for. For example "vault" or "blob".')
+  service: string?
+
+  @description('Required. Resource ID of the subnet where the endpoint needs to be created.')
+  subnetResourceId: string
+
+  @description('Optional. The name of the private DNS zone group to create if privateDnsZoneResourceIds were provided.')
+  privateDnsZoneGroupName: string?
+
+  @description('Optional. The private DNS zone groups to associate the private endpoint with. A DNS zone group can support up to 5 DNS zones.')
+  privateDnsZoneResourceIds: string[]?
+
+  @description('Optional. Custom DNS configurations.')
+  customDnsConfigs: {
+    @description('Required. Fqdn that resolves to private endpoint ip address.')
+    fqdn: string?
+
+    @description('Required. A list of private ip addresses of the private endpoint.')
+    ipAddresses: string[]
+  }[]?
+
+  @description('Optional. A list of IP configurations of the private endpoint. This will be used to map to the First Party Service endpoints.')
+  ipConfigurations: {
+    @description('Required. The name of the resource that is unique within a resource group.')
+    name: string
+
+    @description('Required. Properties of private endpoint IP configurations.')
+    properties: {
+      @description('Required. The ID of a group obtained from the remote resource that this private endpoint should connect to.')
+      groupId: string
+
+      @description('Required. The member name of a group obtained from the remote resource that this private endpoint should connect to.')
+      memberName: string
+
+      @description('Required. A private ip address obtained from the private endpoint\'s subnet.')
+      privateIPAddress: string
+    }
+  }[]?
+
+  @description('Optional. Application security groups in which the private endpoint IP configuration is included.')
+  applicationSecurityGroupResourceIds: string[]?
+
+  @description('Optional. The custom name of the network interface attached to the private endpoint.')
+  customNetworkInterfaceName: string?
+
+  @description('Optional. Specify the type of lock.')
+  lock: lockType
+
+  @description('Optional. Array of role assignments to create.')
+  roleAssignments: roleAssignmentType
+
+  @description('Optional. Tags to be applied on all resources/resource groups in this deployment.')
+  tags: object?
+
+  @description('Optional. Manual PrivateLink Service Connections.')
+  manualPrivateLinkServiceConnections: array?
+
+  @description('Optional. Enable/Disable usage telemetry for module.')
+  enableTelemetry: bool?
+}[]?
+
+type scopedResourceType = {
+  @description('Required. Name of the private link scoped resource.')
+  name: string
+
+  @description('Required. The resource ID of the scoped Azure monitor resource.')
+  linkedResourceId: string
+}[]?
+
+type accessModeType = {
```

Comment:
Thanks for streching already and creating udt for module specific objects. If possible here I'd suggest to go the extra mile and create a type for nested exclusions

Reply:
Why that @eriqua? Technically it does not provide a benefit and in both the ReadMe and JSON it renders the same way. Lastly, we use the same for e.g. DiagnosticSettings. Don't get me wrong, we 'can' do it, but I'd like to understand what the benefit would be as long as the internal type isn't re-used multiple times.

Reply:
Forgot to resolve this one. @ahmadabdalla and I had an offline sync on that and noticed the intellisense works, so I agree, it does not provide any benefit to create one type for each nested level of objects, unless it can be reused in multiple places.

Reply:
Sidenote: We actually do this for PE using RBAC